wolfcrypt/src/aes.c

```diff
@@ -4851,7 +4843,7 @@ static void AesSetKey_C(Aes* aes, const byte* key, word32 keySz, int dir)
     int wc_AesSetKey(Aes* aes, const byte* userKey, word32 keylen,
             const byte* iv, int dir)
     {
-        if (aes == NULL) {
+        if ((aes == NULL) || (userKey == NULL)) {
```

Comment:
I think the check on userKey == NULL might cause issues when using AES struct with hardware keys.

Reply:
Removed check and tests.